test/Microsoft.AspNetCore.Mvc.Core.Test/DependencyInjection/MvcCoreServiceCollectionExtensionsTest.cs

```diff
@@ -104,6 +108,116 @@ public void AddMvcServicesTwice_DoesNotAddDuplicates()
             }
         }
 
+        [Fact]
+        public void AddMvcCore_UsesOriginalPartManager()
+        {
+            // Arrange
+            var manager = new ApplicationPartManager();
+            var services = new ServiceCollection();
+            services.AddSingleton(manager);
+
+            // Act
+            var builder = services.AddMvcCore();
+
+            // Assert
+            // SingleRegistrationServiceTypes_AreNotRegistered_MultipleTimes already checks that no other
+            // ApplicationPartManager (but manager) is registered.
+            Assert.Same(manager, builder.PartManager);
+            Assert.Contains(manager.FeatureProviders, provider => provider is ControllerFeatureProvider);
+        }
+
+        // Regression test for aspnet/Mvc#5554.
+        [Fact]
+        public void AddMvcCore_UsesLastPartManager()
+        {
+            // Arrange
+            var services = new ServiceCollection();
+            var mockManager = new Mock<ApplicationPartManager>(MockBehavior.Strict);
```

Comment:
Why do so many mocks have to be added? Would it be enough to add 1 fake one, then 1 real one, and assert that the 1 real one (which was last) was retrieved?